packages/@vue/cli/lib/options.js

```diff
@@ -5,8 +5,32 @@ const cloneDeep = require('lodash.clonedeep')
 const { error } = require('@vue/cli-shared-utils/lib/logger')
 const { createSchema, validate } = require('@vue/cli-shared-utils/lib/validate')
 
+const xdgConfigPath = () => {
+  const createRcDir = (dir) => {
+    const rcDir = path.join(dir, 'vue')
+    if (!fs.existsSync(rcDir)) {
+      fs.mkdirSync(rcDir, 0o700)
+    }
+    return path.join(rcDir, '.vuerc')
+  }
+
+  let xdgConfigHome = process.env.XDG_CONFIG_HOME
+  if (xdgConfigHome) {
+    return createRcDir(xdgConfigHome)
+  }
+
+  // XDG defined backup folder if XDG_CONFIG_HOME is undefined
+  xdgConfigHome = path.join(os.homedir(), '.config')
```

Comment:
This would be a breaking change. We should only change behavior when `XDG_CONFIG_HOME` is explicitly present.